packages/gatsby-plugin-mdx/gatsby/source-nodes.js

```diff
@@ -133,6 +135,34 @@ module.exports = (
       rawBody: { type: `String!` },
       fileAbsolutePath: { type: `String!` },
       frontmatter: { type: `MdxFrontmatter` },
+      slug: {
+        type: `String`,
+        async resolve(mdxNode, args, context) {
+          try {
+            const nodeWithContext = context.nodeModel.findRootNodeAncestor(
+              mdxNode,
+              node => node.internal && node.internal.type === `File`
+            )
+            const fileRelativePath = nodeWithContext.relativePath
+
+            const parsedPath = path.parse(fileRelativePath)
+
+            let relevantPath
+            if (parsedPath.name === `index`) {
+              relevantPath = fileRelativePath.replace(parsedPath.base, ``)
+            } else {
+              relevantPath = fileRelativePath.replace(parsedPath.ext, ``)
+            }
+            return slugify(relevantPath, {
+              remove: /[^\w\s$*_+~.()'"!\-:@/]/g,
```

Comment:
Why are we removing `-`? That's a pretty common delimiter in URLs

Reply:
I pulled the existing set of ignored characters and added slash. `remove` is a bad signifier for this API, but what it's actually doing is removing that set of characters from the set that the slugify function replaces. So `-` is kept if it's already there. The `my-blog` test case covers that to confirm.